sql/src/main/java/org/apache/druid/sql/avatica/DruidConnection.java

```diff
@@ -63,7 +63,7 @@ public DruidConnection(final String connectionId, final int maxStatements, final
     this.connectionId = Preconditions.checkNotNull(connectionId);
     this.maxStatements = maxStatements;
     this.context = ImmutableMap.copyOf(context);
-    this.statements = new HashMap<>();
+    this.statements = new ConcurrentHashMap<>();
```

Comment:
@zhaojiandong For clarity's sake, could you please change the type of the field to ConcurrentMap, & a comment about why it's both `@GuardedBy` and Concurrent?

```java
// ConcurrentMap even though it is usually accessed synchronized, because it's unsynchronized
// in one case: the onClose function passed into DruidStatements contained by the map. This is
// done to avoid deadlocks; see https://github.com/apache/incubator-druid/issues/6867.
@GuardedBy("statements")
private final ConcurrentMap<Integer, DruidStatement> statements;
```

Reply:
ok, I'll put some comments on statements and onClose, thank for review